src/components/slideshow/index.js

```diff
@@ -0,0 +1,340 @@
+import PropTypes from 'prop-types'
+import React, { PureComponent } from 'react'
+import styled from 'styled-components'
+import get from 'lodash/get'
+import map from 'lodash/map'
+
+const _ = {
+  get,
+  map,
+}
+
+const mockup = {
+  desktop: {
+    container: {
+      width: 759, // px
+    },
+    slide: {
+      width: 687, // px
+      height: 429, // px
+      paddingRight: 4, // px
+    },
+    offset: {
+      left: 30, // px
+    },
+  },
+}
+
+// Assuming there are three images [ A, B, C ] for slideshow.
+// If image B is rendered in the center,
+// users can see part of image A(left side) and image C(right side) with masks.
+// When users click right button to see image C, which means, C is in the center,
+// users still can see part of image B(left side) and image A(right side) with masks.
+//
+// Hence, there are four images rendered arround B at the beginning.
+// The image array should be [ C, A, B, C, A ].
+//
+// `slidesOffset` indicates how many slides rendered before/after image B, which is, 2 (A and C).
+//
+const slidesOffset = 2
+
+// duration of transition of transform(translateX(?px))
+const duration = 300
+
+// current index to indicate which image should be rendered in the center
+const defaultCurIndex = 0
+
+const SlideshowFlexBox = styled.div`
+  max-width: 759px;
+  width: 100%;
+  display: flex;
```

Comment:
It seems that we don't need flexbox for `SlideshowFlexBox`

Reply:
`SlideshowFlexBox` is used to put `SlidesSection`, `PrevNextSection` and `DescriptionSection` in the right position.

`justify-content: space-between` to make `PrevNextSection` in the left corner, and `DescriptionSection` in the right corner.

`SlidesSection > flex-basis: 100%`  -> make it fill the `SlideshowFlexBox`.

`flex-wrap: wrap` -> make `PrevNextSection` and `DescriptionSection` below the `SlidesSection`.